presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ExtractSpatialJoins.java

```diff
@@ -429,6 +445,32 @@ else if (alignment < 0) {
                 kdbTree.map(KdbTreeUtils::toJson)));
     }
 
+    private static boolean isSpatialFunctionOnSphericalGeographyArguments(FunctionCall spatialFunction, Context context, Metadata metadata, SqlParser sqlParser)
```

Comment:
I think this change can be simplified by introducing `getExpressionType(expression, ...)` helper method and adding a check for `first/secondArgument` to `tryCreateSpatialJoinByFunction`:

```
        Type sphericalGeographyType = metadata.getType(SPHERICAL_GEOGRAPHY_TYPE_SIGNATURE);
        if (getExpressionType(firstArgument, context, metadata, sqlParser).equals(sphericalGeographyType)
            || getExpressionType(secondArgument, context, metadata, sqlParser).equals(sphericalGeographyType)) {
            return Result.empty();
        }

    private static Type getExpressionType(Expression expression, Context context, Metadata metadata, SqlParser sqlParser)
    {
        Type type = getExpressionTypes(context.getSession(), metadata, sqlParser, context.getSymbolAllocator().getTypes(), expression, emptyList(), WarningCollector.NOOP)
                .get(NodeRef.of(expression));
        verify(type != null);
        return type;
    }
```